metrics_proto_api_test.go

```diff
@@ -85,6 +86,74 @@ func TestVariousCasesFromFile(t *testing.T) {
 	}
 }
 
+func TestMetricsWithPrefix(t *testing.T) {
+	server, addr, doneFn := createFakeServer(t)
```

Comment:
Nit: looks like these few lines appear multiple times. How about having a func `createFakeServerAndConn` that returns both the server and connection? The returned `doneFn` can incorporate `conn.Close()` too.